packages/kit/src/exports/public.d.ts

```diff
@@ -1493,4 +1520,266 @@ export interface Snapshot<T = any> {
 	restore: (snapshot: T) => void;
 }
 
+/**
+ * The return value of a remote `form` function.
+ * Spread it onto a `<form>` element to connect the form with the remote form action.
+ * ```svelte
+ * <script>
+ *   import { createTodo } from './todos.remote.js';
+ * </script>
+ *
+ * <form {...createTodo}>
+ *   <input type="text" name="name" />
+ *   <!-- ... -->
+ * </form>
+ * ```
+ * Use the `enhance` method to influence what happens when the form is submitted.
+ * ```svelte
+ * <script>
+ *   import { getTodos, createTodo } from './todos.remote.js';
+ * </script>
+ *
+ * <form {...myFormAction.enhance(async ({ data, submit }) => {
+ *   // `data` is an instance of FormData (https://developer.mozilla.org/en-US/docs/Web/API/FormData)
+ *   const text = data.get('text');
+ *   const todo = { text, done: false };
+ *
+ *   // `updates` and `withOverride` enable optimistic UI updates
+ *   await submit().updates(getTodos.withOverride((todos) => [...todos, todo]));
+ * })}>
+ *   <input name="text" />
+ *   <!-- ... -->
+ * </form>
+ *
+ * <ul>
+ * 	{#each await getTodos() as todo}
+ * 		<li>{todo.text}</li>
+ * 	{/each}
+ * </ul>
+ * ```
+ */
+export type RemoteFormAction<Success, Failure> = ((data: FormData) => Promise<void>) & {
```

Comment:
I think we should maybe rename this so that it's easier to distinguish from form actions (which are also 'remote' insofar as they run on the server). Not sure what a better name is though, just leaving a note to self

Reply:
`RemoteFormFunction` since it's "remote functions"?
But also, we shouldn't let us guide too much by the potential confusion, assuming that "old" form actions are deprecated at some point

Reply:
Gonna change it to just `RemoteForm`. I don't think there's any good reason for it to be callable, that would at best encourage bad practices IMHO, so 'Function' would be strange. But an 'Action' that has an `action` property (`RemoteFormAction['action']`) is odd too.

I think we should have `RemoteCommand`, RemoteForm`, `RemoteResource` and `RemoteQuery`, where `prerender` returns a `RemoteResource` and `query` returns a `RemoteQuery` which extends `RemoteResource` with `refresh` and `withOverride`. Will open a PR